controllers/ibmpowervsmachine_controller_test.go

```diff
@@ -0,0 +1,562 @@
+							Kind:       "Cluster",
+							Name:       "capi-test-3",
+							UID:        "1",
+						},
+					},
+				}, Spec: infrav1beta1.IBMPowerVSMachineSpec{
+					ServiceInstanceID: "service-instance-1",
+					ImageRef: &corev1.LocalObjectReference{
+						Name: "capi-image",
+					}},
+			},
+			ownerMachine:   &capiv1beta1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "capi-test-machine"}},
+			ownerCluster:   &capiv1beta1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "capi-test-3"}, Spec: capiv1beta1.ClusterSpec{InfrastructureRef: &corev1.ObjectReference{Name: "powervs-cluster"}}},
+			powervsCluster: &infrav1beta1.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "powervs-cluster"}, Spec: infrav1beta1.IBMPowerVSClusterSpec{ServiceInstanceID: "service-instance-1"}},
+			expectError:    false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			reconciler := &IBMPowerVSMachineReconciler{
+				Client: testEnv.Client,
+				Log:    klogr.New(),
+			}
+			ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("namespace-%s", util.RandomString(5)))
+			g.Expect(err).To(BeNil())
+			defer func() {
+				g.Expect(testEnv.Cleanup(ctx, ns)).To(Succeed())
+			}()
+
+			createObject(g, tc.ownerCluster, ns.Name)
+			defer cleanupObject(g, tc.ownerCluster)
+
+			createObject(g, tc.powervsCluster, ns.Name)
+			defer cleanupObject(g, tc.powervsCluster)
+
+			createObject(g, tc.ownerMachine, ns.Name)
+			defer cleanupObject(g, tc.ownerMachine)
+
+			createObject(g, tc.powervsMachine, ns.Name)
+			defer cleanupObject(g, tc.powervsMachine)
+
+			if tc.powervsMachine != nil {
+				g.Eventually(func() bool {
+					machine := &infrav1beta1.IBMPowerVSMachine{}
+					key := client.ObjectKey{
+						Name:      tc.powervsMachine.Name,
+						Namespace: ns.Name,
+					}
+					err = testEnv.Get(ctx, key, machine)
+					return err == nil
+				}, 10*time.Second).Should(Equal(true))
+
+				_, err := reconciler.Reconcile(ctx, ctrl.Request{
+					NamespacedName: client.ObjectKey{
+						Namespace: tc.powervsMachine.Namespace,
+						Name:      tc.powervsMachine.Name,
+					},
+				})
+				if tc.expectError {
+					g.Expect(err).ToNot(BeNil())
+				} else {
+					g.Expect(err).To(BeNil())
+				}
+			} else {
+				_, err = reconciler.Reconcile(ctx, ctrl.Request{
+					NamespacedName: client.ObjectKey{
+						Namespace: "default",
+						Name:      "test",
+					},
+				})
+				g.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestIBMPowerVSMachineReconciler_Delete(t *testing.T) {
+	var (
+		reconciler   IBMPowerVSMachineReconciler
+		machineScope *scope.PowerVSMachineScope
+		mockCtrl     *gomock.Controller
+		mockpowervs  *mock.MockPowerVS
+	)
+	mockCtrl = gomock.NewController(t)
+	mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	reconciler = IBMPowerVSMachineReconciler{
+		Client: testEnv.Client,
+		Log:    klogr.New(),
+	}
+	g := NewWithT(t)
+	t.Run("Reconciling deleting IBMPowerVSMachine ", func(t *testing.T) {
```

Comment:
can this be converted into tabular format?

Reply:
It could be converted but I found it easier to add test cases for these two functions in non tabular format because it provided flexibility to customise the inputs and for the mock client operations.